src/screens/CustomizedFoodPage.js

```diff
@@ -156,45 +205,133 @@ const populateingredients = (array) => {
         return {
           ...ingredient,
           kcal100: data.parsed[0].food.nutrients.ENERC_KCAL,
+          quantity: 0,
         };
       });
   });
 };
 
 export default function CustomizedFoodPage() {
   const [ingredientsKcal, setIngredientsKcal] = useState([]);
+  const [chosenIngredientsList, setChosenIngredientsList] = useState(
+    initialIngredientsList
+  );
 
   useEffect(() => {
     Promise.all(populateingredients(ingrForRequests)).then((newingredients) => {
       setIngredientsKcal(newingredients);
     });
   }, []);
 
+  const setServingQuantity = (ingredientId) => {
+    const ingredExists = chosenIngredientsList.filter(
+      (ingred) => ingred.id === ingredientId
+    );
+
+    return ingredExists.length > 0 ? (
+      <span>{ingredExists[0].quantity} </span>
+    ) : (
+      <span>0 </span>
+    );
+  };
+
+  const handleChangeQuantity = (id, operator) => {
+    const ingredToUpdate = chosenIngredientsList.filter(
+      (ingred) => ingred.id === id
+    );
+    const currentQuantity =
+      ingredToUpdate.length === 0 ? 0 : ingredToUpdate[0].quantity;
+
+    if (operator === 'add') {
+      if (ingredToUpdate.length === 0) {
+        const ingredToAdd = ingredientsKcal.filter(
+          (ingred) => ingred.id === id
+        );
+        setChosenIngredientsList((IngredientsList) => [
+          ...IngredientsList,
+          {
+            id: ingredToAdd[0].id,
+            name: ingredToAdd[0].name,
+            imgsrc: ingredToAdd[0].imglayer,
+            quantity: 1,
+            serving: ingredToAdd[0].serving,
+            kcal100: ingredToAdd[0].kcal100,
+          },
+        ]);
+      } else {
+        ingredToUpdate[0].quantity += 1;
```

Comment:
mutations are forbidden on react state data

Reply:
prefer mapping over the array in the state to create a new version with the quantity updated

Reply:
or use https://immerjs.github.io/immer/#quick-example